tests/unit/test_template_api.py

```diff
@@ -179,6 +179,16 @@ def test_template_inline_dependencies(bm: BuildingMOTIF):
     }
 
 
+def test_template_inline_dependencies_with_optional(bm: BuildingMOTIF):
+    # fixes https://github.com/NREL/BuildingMOTIF/issues/237
+    lib = Library.load(directory="tests/unit/fixtures/optional-inline")
+    templ = lib.get_template_by_name("hot-water-coil")
+    templ = templ.inline_dependencies()
+    bindings, _ = templ.fill(BLDG, include_optional=False)
+    assert "supply-water-temp" in bindings.keys()
+    assert "name" in bindings.keys()
```

Comment:
Could you add a negative test as well? fill with optional included and assert these keys are not present?

Reply:
Added!